File: stress/t/ctl_statemach.f
      PROGRAM CTLSM
      INTEGER*8 STATE,I,SYM,ACC,STEPS
      STATE=0
      ACC=0
      STEPS=0
      I=0
   10 IF (I.GE.40) GOTO 20
      SYM=MOD(I*3+1,3_8)
      IF (STATE.EQ.0) THEN
        IF (SYM.EQ.0) THEN
          STATE=1
        ELSE
          STATE=0
        END IF
      ELSE IF (STATE.EQ.1) THEN
        IF (SYM.EQ.1) THEN
          STATE=2
        ELSE
          STATE=0
        END IF
      ELSE IF (STATE.EQ.2) THEN
        IF (SYM.EQ.2) THEN
          STATE=0
          ACC=ACC+1
        ELSE
          STATE=1
        END IF
      ELSE
        STATE=0
      END IF
      STEPS=STEPS+1
      I=I+1
      GOTO 10
   20 CONTINUE
      WRITE(*,'(A,I0)') 'accepts=',ACC
      WRITE(*,'(A,I0)') 'state=',STATE
      END

File: stress/t/ctl_statemach.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t state=0,i=0,sym=0,accepts=0,steps=0;
  for(i=0;i<40;i=i+1){
    sym=(i*3+1)%3;
    if(state==0){ if(sym==0){ state=1; } else { state=0; } }
    else { if(state==1){ if(sym==1){ state=2; } else { state=0; } }
      else { if(state==2){ if(sym==2){ state=0; accepts=accepts+1; } else { state=1; } }
        else { state=0; } } }
    steps=steps+1;
  }
  printf("accepts="); printf("%ld",(long)accepts); printf("\n");
  printf("state="); printf("%ld",(long)state); printf("\n");
  return 0;
}
